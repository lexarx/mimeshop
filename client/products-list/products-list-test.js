jest.mock('api-request');
jest.mock('add-to-cart-form/add-to-cart-form', () => () => 'Add to cart form');
import React from 'react'
import ProductsList from 'products-list/products-list'
import { mount } from 'enzyme'
import apiRequest from 'api-request'

test('displays products list', async () => {
	apiRequest.mockImplementation((method, url) => {
		if (method === 'GET' && url === 'products') {
			return [
				{
					id: '1',
					shortDescription: `Short description 1`,
					preview: 'http://via.placeholder.com/100?text=P1',
					price: 100.00
				},
				{
					id: '2',
					shortDescription: `Short description 2`,
					preview: 'http://via.placeholder.com/100?text=P2',
					price: 100.00
				}
			];
		}
	});
	const productsList = mount(<div><ProductsList/></div>);
	await new Promise(resolve => setTimeout(resolve, 0));
	productsList.update();
	expect(productsList).toMatchSnapshot();
});

test('navigates to product page', async () => {
	apiRequest.mockImplementation((method, url) => {
		if (method === 'GET' && url === 'products') {
			return [
				{
					id: '1',
					shortDescription: `Short description 1`,
					preview: 'http://via.placeholder.com/100?text=P1',
					price: 100.00
				}
			];
		}
	});
	const navigate = jest.fn();
	const productsList = mount(<div><ProductsList navigate={navigate}/></div>);
	await new Promise(resolve => setTimeout(resolve, 0));
	productsList.update();
	productsList.find('img').simulate('click');
	expect(navigate).toHaveBeenCalledWith('/products/1');
});