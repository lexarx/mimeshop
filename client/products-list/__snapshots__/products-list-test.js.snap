// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays products list 1`] = `
<div>
  <ProductsList>
    <div
      className="product"
      key="1"
    >
      <img
        className="image"
        onClick={[Function]}
        src="http://via.placeholder.com/100?text=P1"
      />
      <div
        className="shortDescription"
        onClick={[Function]}
      >
        Short description 1
      </div>
      <div
        className="price"
      >
        Price: 
        100
      </div>
      <Component
        productId="1"
      >
        Add to cart form
      </Component>
    </div>
    <div
      className="product"
      key="2"
    >
      <img
        className="image"
        onClick={[Function]}
        src="http://via.placeholder.com/100?text=P2"
      />
      <div
        className="shortDescription"
        onClick={[Function]}
      >
        Short description 2
      </div>
      <div
        className="price"
      >
        Price: 
        100
      </div>
      <Component
        productId="2"
      >
        Add to cart form
      </Component>
    </div>
  </ProductsList>
</div>
`;
